autogpt_platform/backend/backend/blocks/example.py

```diff
@@ -0,0 +1,160 @@
+]
+
+TEST_CREDENTIALS = APIKeyCredentials(
+    id="9191c4f0-498f-4235-a79c-59c0e37454d4",
+    provider="example-provider",
+    api_key=SecretStr("mock-example-api-key"),
+    title="Mock Example API key",
+    expires_at=None,
+)
+TEST_CREDENTIALS_INPUT = {
+    "provider": TEST_CREDENTIALS.provider,
+    "id": TEST_CREDENTIALS.id,
+    "type": TEST_CREDENTIALS.type,
+    "title": TEST_CREDENTIALS.title,
+}
+
+
+class ExampleBlock(Block):
+
+    class ExampleBlockInput(BlockSchema):
+        name: str = SchemaField(description="The name of the example block")
+        greeting: str = SchemaField(description="The greeting to display")
+        is_funny: bool = SchemaField(description="Whether the block is funny")
+        # Only if the block needs credentials
+        credentials: ExampleCredentials = CredentialsField()
+
+    class ExampleBlockOutput(BlockSchema):
+        response: dict[str, Any] = SchemaField(
+            description="The response object generated by the example block."
+        )
+        error: str = SchemaField(description="The error from the example block")
+
+    def __init__(self):
+        super().__init__(
+            # The unique identifier for the block, this value will be persisted in the DB.
+            # It should be unique and constant across the application run.
+            # Use the UUID format for the ID.
+            id="380694d5-3b2e-4130-bced-b43752b70de9",
+            # The description of the block, explaining what the block does.
+            description="The example block",
+            # The list of contributors who contributed to the block.
+            # Each contributor is represented by a ContributorDetails object.
+            contributors=[ContributorDetails(name="Craig Swift")],
+            # The set of categories that the block belongs to.
+            # Each category is an instance of BlockCategory Enum.
+            categories={BlockCategory.BASIC},
+            # The schema, defined as a Pydantic model, for the input data.
+            input_schema=self.ExampleBlockInput,
+            # The schema, defined as a Pydantic model, for the output data.
+            output_schema=self.ExampleBlockOutput,
+            # The list or single sample input data for the block, for testing.
+            # This is an instance of the Input schema with sample values.
+            test_input=self.ExampleBlockInput(
+                name="Craig", greeting="Hello", is_funny=True
+            ).model_dump(),
+            # The list or single expected output if the test_input is run.
+            # Each output is a tuple of (output_name, output_data).
+            test_output=[
+                ("response", {"message": "Hello, world!"}),
+            ],
+            # Function names on the block implementation to mock on test run.
+            # Each mock is a dictionary with function names as keys and mock implementations as values.
+            test_mock={
```

Comment:
It’s afaik not possible to mock some calls so a lot of the time static functions are made as part of the block for easy testing to mock everything up at once

```
@staticmethod
def _request_horse(client, name)-> Whatever
    pass
```